cmd/traffic_layout/traffic_layout.cc

```diff
@@ -118,35 +118,43 @@ info(int argc, const char **argv)
 int
 traffic_runroot(int argc, const char **argv)
 {
-  // runroot engine for operations
-  RunrootEngine engine;
-  engine._argc = argc;
-
-  int i = 0;
-  while (argv[i]) {
-    engine._argv.push_back(argv[i]);
-    ++i;
-  }
-  // parse the command line & put into global variable
-  if (!engine.runroot_parse()) {
-    engine.runroot_help_message(true, true);
+  try {
+    // runroot engine for operations
+    RunrootEngine engine;
+    engine._argc = argc;
```

Comment:
Not really related to this PR, but why does `engine` have `_argc` and a `vector` for `argv`? Isn't `_argc` just `_argv.size()`? Why have them separate? Relatedly, it seems wrong to trust `argc` for the count but not use it when populating `_argv`.

Reply:
It was a how @chitianhao wrote it. I agree probably not needed. He can fix this in the next set of improvments